core/database/db.py

```diff
@@ -19,25 +19,44 @@
     dynamodb = boto3.client('dynamodb')
 
 
-# Shared Functions
+## Shared Functions
 
 def delete_item(pk, sk):
     query = {
         "pk": {"S": pk},
         "sk": {"S": sk}
     }
 
-    response = dynamodb.delete_item(TableName=table,
-                                    Key=query)
+    response = dynamodb.delete_item(
+        TableName=table,
+        Key=query
+    )
 
     return response["ResponseMetadata"]["HTTPStatusCode"] == 200
 
 
 def delete_partition(pk):
     block = get_entire_partition(pk)
     for item  in block:
-        delete_item(item["pk"], item["sk"])
-    return
+        response = delete_item(item["pk"], item["sk"])
+        if not response:
+            return False
+
+    return True
+
+
+# Deletes any entry with substr in an entry's sk for the entire database
+def delete_scan(substr):
+    matches = scan_sk_for_substr(substr)
+    if not matches:
+        return False
+
+    for item in matches:
+        response = delete_item(item['sk'], item['sk'])
```

Comment:
Can we put some logging in here for? If you just do something like `print(json.dumps(item)` that should be enough. Print statements are outputted to the CloudWatch logs. Piggybacking on my earlier comment, it would be helpful if part of a user delete fails to be able to pinpoint exactly what got deleted. This is mostly an issue for "scan" since something could match without us intending it to. Delete partition doesn't need it I dont think, but I'm open to your thoughts

Reply:
Yeah I can put that in! That's good to know